modules/transaction-payment/src/lib.rs

```diff
@@ -284,20 +280,33 @@ pub mod module {
 		Multiplier::saturating_from_integer(1)
 	}
 
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The swap path is invalid
+		InvalidSwapPath,
+	}
+
 	/// The next fee multiplier.
 	///
 	/// NextFeeMultiplier: Multiplier
 	#[pallet::storage]
 	#[pallet::getter(fn next_fee_multiplier)]
 	pub type NextFeeMultiplier<T: Config> = StorageValue<_, Multiplier, ValueQuery, DefaultFeeMultiplier>;
 
+	/// TODO: remove it after migrate
 	/// The default fee currency for accounts.
 	///
 	/// DefaultFeeCurrencyId: AccountId => Option<CurrencyId>
 	#[pallet::storage]
 	#[pallet::getter(fn default_fee_currency_id)]
 	pub type DefaultFeeCurrencyId<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, CurrencyId, OptionQuery>;
 
+	/// The alternative fee swap path of accounts.
+	#[pallet::storage]
+	#[pallet::getter(fn alternative_fee_swap_path)]
+	pub type AlternativeFeeSwapPath<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, Vec<CurrencyId>, OptionQuery>;
```

Comment:
`Twox64Concat` might not be safe enough for `T::AccountId`?

For swap path, we might need bounded vec instead.

Reply:
People can only set swap path for themself so it should be safe.
Can hardcode the limit to 3 elements for swap path.